src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +32,35 @@
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${activeTodo.id}`}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => setActiveTodo(null)}
+            />
           </header>
 
           <div className="modal-card-body">
             <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+              {activeTodo.title}
             </p>
 
             <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+              {activeTodo.completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
 
               {' by '}
 
-              <a href="mailto:Sincere@april.biz">Leanne Graham</a>
+              {todoUser && (
+                <a href={`mailto:${todoUser.email}`}>{todoUser.name}</a>
```

Comment:
You're generating a new function on each render by using an inline function for the `href` attribute. It's better to avoid generating new functions during render because it can lead to unnecessary re-renders and negatively impact performance. Consider moving this function outside of the component or memoizing it with `useCallback`.